feeds/wlan-ap/opensync/src/platform/openwrt/src/rrm/src/rrm_channel.c

```diff
@@ -103,6 +101,56 @@ void get_channel_bandwidth(const char* htmode, int *channel_bandwidth)
 	else if(!strcmp(htmode, "HT80"))
 		*channel_bandwidth=80;
 }
+/*
+ * A simple average is calculated against the Noise floor samples.
+ * - Returns a zero, if there are not enough samples in the list.
+ * - Or, returns a calculated avg of the noise samples.
+ * - A circular buffer is considered for storing the samples. New elements replace the
+ * old ones in this buffer.
+ * - The number of samples required is an integer value calculated based on the
+ * configuration time and the sample time.
+ * Example: If the config_time is 120 sec, and sample time is 15 sec, then number of
+ * samples required for averaging is 120/15 = 8 samples.
+ */
+int rrm_calculate_avg_noise_floor(rrm_entry_t *rrm_data, int nf, int config_time, int sample_time)
+{
+	int ii;
+	double avg_mW = 0;
+	int num_samples = config_time/sample_time;
+
+	if (num_samples >= RRM_MAX_NF_SAMPLES)
+		num_samples = RRM_MAX_NF_SAMPLES;
+
+	/* Replace the oldest element in the list with the new element */
+	rrm_data->rrm_chan_nf_samples[rrm_data->rrm_chan_nf_next_el] = nf;
+
```

Comment:
I propose converting to mW before storing in the array.  Then the calculation is easier and we just convert on the return value.

Reply:
Ok. I will make the change.